lib/jsonapi/utils/response/formatters.rb

```diff
@@ -96,9 +106,18 @@ def build_response_document(object, options)
 
           if object.respond_to?(:to_ary)
             records = build_collection(object, options)
-            results.add_result(JSONAPI::ResourcesOperationResult.new(:ok, records, result_options(object, options)))
+
+            if options[:source_resource].present? && options[:relationship_type].present?
+              results.add_result(JSONAPI::RelatedResourcesOperationResult.new(:ok,
+                                                          options[:source_resource],
+                                                          options[:relationship_type],
+                                                          records,
+                                                          result_options(object, options)))
+            else
+              results.add_result(JSONAPI::ResourcesOperationResult.new(:ok, records, result_options(object, options)))
+            end
           else
-            record = turn_into_resource(object, options)
+            record = turn_into_resource(object, options)         
```

Comment:
I don't know if this is necessary. In _jsonapi-resources_ the special handling is only used for `has_many` relationships, `has_one` relationships are handled and rendered the same as regular show requests, i.e., the links in the response is to the show page of an object and not the relationships link.
This makes sense as only for the `has_many` relationship an issue with paging would arise.